app/org/sagebionetworks/bridge/models/schedules/BaseTaskScheduler.java

```diff
@@ -0,0 +1,82 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import static org.sagebionetworks.bridge.models.schedules.ScheduleType.ONCE;
+
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalTime;
+import org.sagebionetworks.bridge.BridgeUtils;
+
+abstract class BaseTaskScheduler implements TaskScheduler {
+
+    protected final String schedulePlanGuid;
+    protected final Schedule schedule;
+    
+    BaseTaskScheduler(String schedulePlanGuid, Schedule schedule) {
+        this.schedulePlanGuid = schedulePlanGuid;
+        this.schedule = schedule;
+    }
+    
+    protected DateTime getStartTimeBasedOnEvent(Schedule schedule, Map<String, DateTime> events) {
+        // If no event is specified, it's enrollment by default.
+        String eventId = schedule.getEventId();
```

Comment:
This is not the only event or possibly even the most important event that can be scheduled against. Scheduling against the last scheduled task is an internal event that we will care about, but researchers will probably schedule against the completion of specific tasks, so event IDs are going to be a combination of the event type, and an identifier for an activity, or a special event like enrollment (as if the first task a user completes is enrolling themselves).